### PR TITLE
BDOG-1268: Upgrade to Play 28

### PR DESCRIPTION
**Pull request by cjwebb**
Also, remove some unneeded dependencies

### CODE REVIEW

File: it/uk/gov/hmrc/fileupload/GetEnvelopeIntegrationSpec.scala [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package uk.gov.hmrc.fileupload
 
 import org.joda.time.DateTime
```
Comment (colin-lamed) [inline it/uk/gov/hmrc/fileupload/GetEnvelopeIntegrationSpec.scala:19]: Can we replace joda DateTime with java.time.Instant?

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -4,31 +4,23 @@ private object AppDependencies {
...
-    "uk.gov.hmrc"              %% "mongo-lock"                % "6.23.0-play-27",
+    "uk.gov.hmrc"              %% "mongo-lock"                % "7.0.0-play-28",
     PlayImport.ws,
-    "uk.gov.hmrc"              %% "bootstrap-backend-play-27" % playBootstrapVersion,
-    "uk.gov.hmrc"              %% "domain"                    % domainVersion,
-    "com.typesafe.akka"        %% "akka-actor"                % akkaVersion,
-    "com.typesafe.akka"        %% "akka-testkit"              % akkaVersion,
-    "org.typelevel"            %% "cats-core"                 % "2.2.0",
+    "uk.gov.hmrc"              %% "bootstrap-backend-play-28" % "4.1.0",
     "org.reactivemongo"        %% "reactivemongo-iteratees"   % "0.18.8",
```
Comment (colin-lamed) [inline project/AppDependencies.scala:13]: Worth switching from reactivemongo to hmrc-mongo. Would be interesting to try out `akka-stream-alpakka-mongodb`

**Review by colin-lamed — APPROVED** [review]

**Comment by cjwebb** [comment]
I'm going to merge, and deploy this version.... just to reduce the scope of future incremental changes. Notably, hmrc-mongo.

File: project/AppDependencies.scala (hunk 2 above)
Comment (cjwebb) [reply project/AppDependencies.scala:13]: alpakka isn't using akka 2.6 yet :(
Looking at the code though, the Mongo version is quite minimal, and we don't need it IMO:
`Source.fromPublisher` takes an Observable, which is what hmrc-mongo already gives us.

Comment (cjwebb) [reply project/AppDependencies.scala:13]: Will be switching to hmrc-mongo in a later PR.

**Pull request merged**